frontend/middleware.ts

```diff
@@ -0,0 +1,15 @@
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+
+// This function can be marked `async` if using `await` inside
+export async function middleware(req: NextRequest) {
+  const token = req.cookies.get('token')?.value
+
+  if (!token) return NextResponse.redirect(new URL('/login', req.url))
+
+  return NextResponse.next()
+}
+
+export const config = {
+  matcher: ['/dashboard']
```

Comment:
Hmmm is there a better way of doing this? IMO this is a bit brittle, because this means that you need to make sure to remember to update this file whenever you add or delete routes that need auth checking. 

What if instead of that, you declare that authentication in the actual page that needs it? 

Have you tried this? https://next-auth.js.org/